test/packetimpact/testbench/connections.go

```diff
@@ -95,6 +97,11 @@ func NewTCPIPv4(t *testing.T, dut DUT, outgoingTCP, incomingTCP TCP) TCPIPv4 {
 	if err != nil {
 		t.Fatalf("can't pick a port: %s", err)
 	}
+
+    if(SetSameLocalRemoteIP == 1) {
+        *localIPv4 = *remoteIPv4
+    }
```

Comment:
This modifies the flag, which we shouldn't do because it would potentially affect future tests that are in this file. 
 Flags should be treated as if they are constants.

Instead of doing all this, can we refactor the `TCPIPv4.Send()` function in [connections.go](https://github.com/google/gvisor/blob/fed59953aad40d89730ebfc6f33b17116c42abcf/test/packetimpact/testbench/connections.go#L153)?  We can split it into two functions, one which has the same name and args and the other which will have signature `func (conn *TCPIPv4) SendFrame(frame Layers)`.  The former will build the `frame` and the latter will do the rest of what the `Send` function already does.

Then, in the test, you can call the SendFrame function directly and put in whatever IP address you want.  You can get the IP address that you want to use from the `TCPIPv4` struct, it's in the `outgoing` member.  You'll need to make `outgoing` exported by changing the name to `Outgoing`.

Reply:
[GES] As suggested, we will avoid usage of global variable. Will add new function SendFrame(..) in connections.go and pass parameters tcp_flag, local_ip, remote_ip as parameter And same will be invoked in the test code
Existing functions NewTCPIPv4(..) and Send(..) in connections.go will not be modified

Reply:
`SendFrame(frame []Layers)`

There should be just one param and it is the complete frame to send.  The existing `Send(...)` function can build a frame and then call `SendFrame`

Reply:
[GES] Ok. Will take care